### PR TITLE
volume: acquire the lock before list plugins that have volume limits on nodes to avoid data race

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
volume: acquire the lock before list plugins that have volume limits on nodes to avoid data race

**Which issue(s) this PR fixes**:

Ref #90540

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: pkg/volume/plugins.go [hunk 1]
```diff
@@ -750,6 +750,9 @@ func (pm *VolumePluginMgr) refreshProbedPlugins() {
 
 // ListVolumePluginWithLimits returns plugins that have volume limits on nodes
 func (pm *VolumePluginMgr) ListVolumePluginWithLimits() []VolumePluginWithAttachLimits {
+	pm.mutex.Lock()
+	defer pm.mutex.Unlock()
```
Comment (gaurav1086) [inline pkg/volume/plugins.go:754]: How about use only read lock ?

pm.mutex.RLock()
defer pm.mutex.RUnlock()

Comment (SataQiu) [reply pkg/volume/plugins.go:754]: Emm... But `sync.Mutex` only supports `Lock`. Maybe you thought it as `sync.RWMutex` ?
```go
// VolumePluginMgr tracks registered plugins.
type VolumePluginMgr struct {
	mutex         sync.Mutex
	plugins       map[string]VolumePlugin
	prober        DynamicPluginProber
	probedPlugins map[string]VolumePlugin
	Host          VolumeHost
}
```

**Pull request closed without merging**